sherpa/data.py

```diff
@@ -490,19 +490,31 @@ class DataSimulFit(Data):
 
     """
 
-    def __init__(self, name, datasets):
+    def __init__(self, name, datasets, numcores=1):
```

Comment:
Given that `numcores` is only used in `eval_model_to_fit`, it looks like it would be better if it were an argument to `eval_model_to_fit` rather than an attribute of the `BaseData` object. Or does the "complicated" way we implement/use `BaseData` mean that this doesn't work?

Reply:
For now, let's leave it the constructor as is for testing purposes. The default value for numcores=1; However, if you want to run with multicores, say 3, then run your script with
```
fit(numcores=3)
```
I tested this claim by adding a printing statement in the constructor of the DataSimulFit
```
(sherpa3) [dtn@devel12 sherpa]$ git diff sherpa/data.py
diff --git a/sherpa/data.py b/sherpa/data.py
index de2d0302..701c2eb2 100644
--- a/sherpa/data.py
+++ b/sherpa/data.py
@@ -491,6 +491,7 @@ class DataSimulFit(Data):
     """
 
     def __init__(self, name, datasets, numcores=1):
+        print('DataSimulFit::__init__ numcores =', numcores)
         if len(datasets) == 0:
             raise DataErr('zerodatasimulfit', type(self).__name__)
         datasets = tuple(datasets)
```

(sherpa3) [dtn@devel12 integrate]$ python fit_3_spectra_simple.py | grep numbers
DataSimulFit::__init__ numcores = 1

Changing the script from fit() to fit(numcores=3)

(sherpa3) [dtn@devel12 integrate]$ diff fit_3_spectra_simple.py tmp.py
68d67
<
85c84
<     fit()
---
>     fit(numcores=3)

(sherpa3) [dtn@devel12 integrate]$ python tmp.py | grep numcores
DataSimulFit::__init__ numcores = 3

Reply:
For fun, I moved the ```numcores``` from the constructor of the DataSimulFit class and moved it to the ```eval_model_to_fit```  as Doug suggested.  Unfortunately, the ```fit(numcores=3)``` has no effect.  Not a good idea in my opinion, I'm willing to move it but I suspect the user interface is not going to be pretty but that is just my opinion